packages/typescript-eslint/src/config-helper.ts

```diff
@@ -85,11 +85,30 @@ export interface ConfigWithExtends extends TSESLint.FlatConfig.Config {
 export function config(
   ...configs: ConfigWithExtends[]
 ): TSESLint.FlatConfig.ConfigArray {
-  return configs.flatMap(configWithExtends => {
+  return configs.flatMap((configWithExtends, configIndex) => {
     const { extends: extendsArr, ...config } = configWithExtends;
     if (extendsArr == null || extendsArr.length === 0) {
       return config;
     }
+    const undefinedExtensions = extendsArr.reduce<number[]>(
+      (acc, extension, extensionIndex) => {
+        const maybeExtension = extension as
+          | TSESLint.FlatConfig.Config
+          | undefined;
+        if (!maybeExtension) {
+          acc.push(extensionIndex);
+        }
+        return acc;
+      },
+      [],
+    );
+    if (undefinedExtensions.length) {
+      throw new Error(
+        `Your config at index ${configIndex} contains undefined extensions ` +
+          `at the following indices: ${undefinedExtensions.join(', ')}.\n` +
+          'This is likely due to a problem with extension import paths.',
```

Comment:
Sorry, just thinking on this now: I don't know that we can make this conclusion. Sometimes users use the wrong properties, or have the wrong version installed, or a myriad of other things. I'd just remove the line and let them investigation on their own:

```suggestion
```